client/src/main/java/io/pravega/schemaregistry/client/SchemaRegistryClientImpl.java

```diff
@@ -486,4 +507,40 @@ public void close() throws Exception {
             client.close();
         }
     }
+
+    private SSLContext getSSLContext(SchemaRegistryClientConfig config) {
+        try {
+            KeyStore trustStore;
+            if (config.isSystemPropTls()) {
+                return SSLContext.getDefault();
+            } else if (config.isCertificateTrustStore()) {
+                trustStore = CertificateUtils.createTrustStore(config.getTrustStore());
+            } else {
+                trustStore = getTrustStore(config);
+            }
+            TrustManagerFactory factory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            factory.init(trustStore);
+            SSLContext tlsContext = SSLContext.getInstance(TLS);
+            tlsContext.init(null, factory.getTrustManagers(), null);
+            return tlsContext;
+        } catch (KeyManagementException | KeyStoreException | NoSuchAlgorithmException |
+                CertificateException | IOException e) {
+            throw new IllegalArgumentException("Failure initializing trust store", e);
```

Comment:
`IllegalArgumentException` appears to be a wrong characterization at this point.

Reply:
illegal argument because except IO exception, all other exceptions are representative of argument being incorrect.

Reply:
what do you suggest should be the appropriate exception though?

Reply:
Maybe `IllegalStateException` is a better characterization here. Even a custom Schema Registry exception'd work better here. Also, we could perform input validation upfront and throw `IllegalArgumentException` eagerly.

Reply:
I will change it to IllegalStateException.